flink-core/src/main/java/org/apache/flink/metrics/reporter/AbstractReporter.java

```diff
@@ -32,9 +35,11 @@
  */
 @PublicEvolving
 public abstract class AbstractReporter implements MetricReporter {
+	protected final Logger log = LoggerFactory.getLogger(getClass());
```

Comment:
Because subclasses should be able to have their own logger which is instantiated with the correct class name.